tests/src/core/testqgis.cpp

```diff
@@ -342,8 +342,13 @@ void TestQgis::qVariantCompare_data()
   QTest::newRow( "qstringlist 5" ) << QVariant( QStringList() << QStringLiteral( "aa" ) ) << QVariant( QStringList() << QStringLiteral( "aa" ) << QStringLiteral( "xx" ) ) << true << false << -1;
   QTest::newRow( "qstringlist 6" ) << QVariant( QStringList() << QStringLiteral( "aa" ) << QStringLiteral( "xx" ) ) << QVariant( QStringList() << QStringLiteral( "aa" ) ) << false << true << 1;
   QTest::newRow( "string both empty" ) << QVariant( QString() ) << QVariant( QString() ) << false << false << 0;
+#if QT_VERSION >= QT_VERSION_CHECK( 6, 0, 0 )
+  QTest::newRow( "string" ) << QVariant( "a b c" ) << QVariant( "d e f" ) << true << false << -3;
```

Comment:
I'd lean toward modifying qVariantCompare to always return -1 / +1 instead -- that'd give more predictable behavior across versions, and I can't see a strong need for non -1/0/+1 values.